File: Python/BinaryTreeInorderTraversal.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def inorderTraversal(root):
        result = []

        def helper(node):
            if node is None:
                return
            
            helper(node.left)
            result.append(node.val)
            helper(node.right)


        helper(root)
        
        return result

File: Javascript/BinaryTreeInorderTraversal.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0 : val
    this.left = (left === undefined) ? null : left
    this.right = (right === undefined) ? null : right
}

var inorderTraversal = function (root) {

    let result = []

    function helper(node){
        if(node === null){
            return
        }

        helper(node.left)
        result.push(node.val)
        helper(node.right)
    }

    inorderTraversal(root)
    return result

}